From 61b9d2b3a5be74ea5f5140a18f93ea54a01e3fc3 Mon Sep 17 00:00:00 2001
From: Brian Bockelman <bbockelm@cse.unl.edu>
Date: Wed, 25 Mar 2020 21:49:24 -0500
Subject: [PATCH] Fix MKCOL response when we have an EEXIST.

Per the WebDAV spec, the correct MKCOL response for when a collection
already exists is 405.

Without this fix, `gfal-mkdir` interprets this as a file-not-found.
---
 src/XrdHttp/XrdHttpReq.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/XrdHttp/XrdHttpReq.cc b/src/XrdHttp/XrdHttpReq.cc
index 92dc45d2667..f86ceb52f73 100644
--- a/xrootd/src/XrdHttp/XrdHttpReq.cc
+++ b/xrootd/src/XrdHttp/XrdHttpReq.cc
@@ -943,6 +943,9 @@ void XrdHttpReq::mapXrdErrorToHttpStatus() {
       case kXR_isDirectory:
         httpStatusCode = 409; httpStatusText = "Resource is a directory";
         break;
+      case kXR_InvalidRequest:
+        httpStatusCode = 405; httpStatusText = "Method is not allowed";
+        break;
       default:
         break;
     }
